### PR TITLE
Change error packages for events 

### PR DESCRIPTION
**Pull request by s-christoff**
Currently using Boundary errors package in the event package would cause a circular dependency for future work. Changing this to std error package.

### CODE REVIEW

File: internal/observability/event/context.go [hunk 1]
```diff
@@ -118,29 +115,25 @@ func WriteError(ctx context.Context, caller Op, e error, opt ...Option) {
 	if !ok {
 		eventer = SysEventer()
 		if eventer == nil {
-			logger := hclog.New(nil)
-			logger.Error(fmt.Sprintf("%s: no eventer available to write error: %s", op, e))
+			fmt.Errorf("%s: no eventer available to write error: %s", op, e)
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/observability/event/context.go:118]: you need to log this... why did you remove the logger code?

File: internal/observability/event/context.go [hunk 2]
```diff
@@ -118,29 +115,25 @@ func WriteError(ctx context.Context, caller Op, e error, opt ...Option) {
...
 			return
 		}
 	}
 	opts := getOpts(opt...)
 	if opts.withRequestInfo == nil {
 		var err error
 		if opt, err = addCtxOptions(ctx, opt...); err != nil {
-			eventer.logger.Error(errors.Wrap(err, op).Error())
-			eventer.logger.Error(fmt.Sprintf("%s: unable to process context options to write error: %s", op, e))
+			fmt.Errorf("%s: unable to process context options to write error: %w", op, e)
```
Comment (jimlambrt) [inline internal/observability/event/context.go:126]: you need to log this... why did you remove the logger code?

File: internal/observability/event/context.go [hunk 3]
```diff
@@ -118,29 +115,25 @@ func WriteError(ctx context.Context, caller Op, e error, opt ...Option) {
...
 			return
 		}
 	}
 	ev, err := newError(caller, e, opt...)
 	if err != nil {
-		eventer.logger.Error(errors.Wrap(err, op).Error())
-		eventer.logger.Error(fmt.Sprintf("%s: unable to create new error to write error: %s", op, e))
+		fmt.Errorf("%s: unable to create new error to write error: %w", op, e)
```
Comment (jimlambrt) [inline internal/observability/event/context.go:132]: you need to log this... why did you remove the logger code?

File: internal/observability/event/context_test.go [hunk 4]
```diff
@@ -53,27 +52,27 @@ func Test_NewRequestInfoContext(t *testing.T) {
...
 			wantErrContains: "missing request info",
 		},
 		{
 			name:            "missing-request-info-id",
 			ctx:             context.Background(),
 			requestInfo:     testInfoMissingId,
-			wantErrMatch:    errors.T(errors.InvalidParameter),
-			wantErrContains: "missing request info id",
+			wantErrIs:       event.ErrInvalidParameter,
+			wantErrContains: "missing request",
```
Comment (jimlambrt) [inline internal/observability/event/context_test.go:75]: why has wantErrContains changed?

File: internal/observability/event/event_audit.go [hunk 5]
```diff
@@ -98,31 +97,32 @@ func (a *audit) FlushEvent() bool {
...
+			return "", nil, fmt.Errorf("%s: event %d has an invalid id: %s != %s: %w", op, i, gated.Id, validId, ErrInvalidParameter)
 		}
 		if gated.Version != auditVersion {
-			return "", nil, errors.New(errors.InvalidParameter, op, fmt.Sprintf("event %d has an invalid version: %s != %s", i, gated.Version, auditVersion))
+			return "", nil, fmt.Errorf("%s: event %d has an invalid version: %s != %s: %w", op, i, gated.Version, auditVersion, ErrInvalidParameter)
 		}
 		if gated.Type != string(apiRequest) {
-			return "", nil, errors.New(errors.InvalidParameter, op, fmt.Sprintf("event %d has an invalid type: %s != %s", i, gated.Type, string(AuditType)))
+			return "", nil, fmt.Errorf("%s: event %d has an invalid type: %s != %s: %w", op, i, gated.Type, string(AuditType), ErrInvalidParameter)
+
```
Comment (jimlambrt) [inline internal/observability/event/event_audit.go:125]: why the added empty line?  perhaps run `make fmt` on this PR?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
Great work!  Just a few things, and in particular the code that falls back to a hclog.Logger when no eventer is available.

File: internal/observability/event/context_test.go (hunk 4 above)
Comment (s-christoff) [reply internal/observability/event/context_test.go:75]: whoops

File: internal/observability/event/context.go [hunk 6]
```diff
@@ -119,28 +118,26 @@ func WriteError(ctx context.Context, caller Op, e error, opt ...Option) {
...
+			eventer.logger.Error(fmt.Sprintf("%s: %v", op, err))
+			eventer.logger.Error(fmt.Sprintf("%s: unable to process context options to write error: %v", op, e))
 			return
 		}
 	}
 	ev, err := newError(caller, e, opt...)
 	if err != nil {
-		eventer.logger.Error(errors.Wrap(err, op).Error())
-		eventer.logger.Error(fmt.Sprintf("%s: unable to create new error to write error: %s", op, e))
+		eventer.logger.Error(fmt.Sprintf("%s: unable to create new error to write error: %v", op, e))
```
Comment (jimlambrt) [inline internal/observability/event/context.go:136]: I think, you've dropped the err from newError(...) here.  perhaps we still want to log that as well.

File: internal/observability/event/context.go [hunk 7]
```diff
@@ -119,28 +118,26 @@ func WriteError(ctx context.Context, caller Op, e error, opt ...Option) {
...
 		return
 	}
 	if err := eventer.writeError(ctx, ev); err != nil {
-		eventer.logger.Error(errors.Wrap(err, op).Error())
-		eventer.logger.Error(fmt.Sprintf("%s: unable to write error: %s", op, e))
+		eventer.logger.Error(fmt.Sprintf("%s: unable to write error: %v", op, e))
```
Comment (jimlambrt) [inline internal/observability/event/context.go:140]: I think, you've dropped the err from writeError(...) here.  perhaps we still want to log that as well.

**Review by jimlambrt — CHANGES_REQUESTED** [review]
just a few more things...

**Review by jimlambrt — APPROVED** [review]
ty!
